expression/builtin_time.go

```diff
@@ -2867,6 +2867,13 @@ func (du *baseDateArithmitical) add(ctx sessionctx.Context, date types.Time, int
 		date.SetFsp(6)
 	}
 
+	// fix https://github.com/pingcap/tidb/issues/11329
+	if goTime.Year() == 0 {
+		hour, minute, second := goTime.Clock()
+		date.SetCoreTime(types.FromDate(0, 0, 0, hour, minute, second, goTime.Nanosecond()<<3))
```

Comment:
`goTime.Nanosecond()<<3`  ?  Should it be `goTime.Nanosecond()/1000`?

Reply:
I think they have same result, and bitwise operations are faster than divide.

Reply:
`goTime.Nanosecond()<<3` means  `goTime.Nanosecond()*8`

Reply:
Sorry, It's my mistake, I'll fix it right now.